49 plant selector search and add frontend integration with backend

jiayi-jenny-zhou opened the pull request.

kev1n commented:
<img width="358" alt="image" src="https://github.com/user-attachments/assets/788313ab-c5d0-4f8f-8388-bb259e73d0ce" />
Not a fan of this view. Feels extremely cluttered.

"Editing Plant" should become "Edit Plant"

We don't want delete to be green, because they'll most likely click it.

In any case, we should be matching the figma for design. Make sure to follow it when you're making future PRs, and also fix it in this to adhere to it. Also, going through the creating new observation flow, there are many fields that have the red required asterisk next to them when not all of them are actually required. See the spec.

kev1n reviewed (changes requested):
See comments

kev1n commented:
Fix merge conflict as well

jiayi-jenny-zhou commented:
![image](https://github.com/user-attachments/assets/8fce961b-215c-4c3e-9463-39f4e96fd95b)

jiayi-jenny-zhou commented:
All required fields have been fixed

jiayi-jenny-zhou commented:
post-review commit summary:
- "forb" and "uncategorized" have been added to the database and frontend
- Fixed nested form bug: linked dialog closings/submits due to nested submit buttons, have to manually validate/error on plantinfo level
- Added delete functionality and additional display details to plantinfo searchbar
- minor backend bugfixes

As asked:
- Realigned requirements
- Added svgs for pen and trashcan for less clutter

kev1n commented on apps/frontend/public/icons/pen.svg:
Ideally we would use lucide-react to import our icons and svgs, though we're tight on time so I won't be so stringent
